rustler_codegen/src/encode_decode_templates.rs

```diff
@@ -31,25 +32,88 @@ pub(crate) fn decoder(ctx: &Context, inner: TokenStream) -> TokenStream {
         let where_clause = impl_generics.make_where_clause();
 
         for lifetime in lifetimes {
-            let mut puncated = syn::punctuated::Punctuated::new();
-            puncated.push(lifetime.clone());
+            let mut punctuated = syn::punctuated::Punctuated::new();
+            punctuated.push(lifetime.clone());
             let predicate = syn::PredicateLifetime {
                 lifetime: decode_lifetime.clone(),
                 colon_token: syn::token::Colon {
                     spans: [Span::call_site()],
                 },
-                bounds: puncated,
+                bounds: punctuated,
             };
             where_clause.predicates.push(predicate.into());
 
-            let mut puncated = syn::punctuated::Punctuated::new();
-            puncated.push(decode_lifetime.clone());
+            let mut punctuated = syn::punctuated::Punctuated::new();
+            punctuated.push(decode_lifetime.clone());
             let predicate = syn::PredicateLifetime {
                 lifetime: lifetime.clone(),
                 colon_token: syn::token::Colon {
                     spans: [Span::call_site()],
                 },
-                bounds: puncated,
+                bounds: punctuated,
+            };
+            where_clause.predicates.push(predicate.into());
+        }
+    }
+
+    let type_parameters: Vec<_> = generics
```

Comment:
What kind of generics do we need to skip here? That's not clear from this source location alone. A comment might help.

Reply:
I added a comment, does it help?

Reply:
Yep, thanks.